src/test/java/it/com/gradle/develocity/bamboo/ShortLivedTokenClientTest.java

```diff
@@ -0,0 +1,109 @@
+package it.com.gradle.develocity.bamboo;
+
+import com.gradle.develocity.bamboo.DevelocityAccessCredential;
+import com.gradle.develocity.bamboo.ShortLivedTokenClient;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.NullAndEmptySource;
+import org.junit.jupiter.params.provider.ValueSource;
+import ratpack.test.embed.EmbeddedApp;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.isEmptyOrNullString;
+import static org.hamcrest.Matchers.not;
+
+public class ShortLivedTokenClientTest implements AfterEachCallback {
+
+    private final ShortLivedTokenClient shortLivedTokenClient = new ShortLivedTokenClient();
+
+    private EmbeddedApp mockDevelocityServer;
+
+    @Override
+    public void afterEach(ExtensionContext context) {
+        if (mockDevelocityServer != null) {
+            mockDevelocityServer.close();
+        }
+    }
+
+    @ParameterizedTest
+    @NullAndEmptySource
+    @ValueSource(strings = {"2"})
+    void shortLivedTokenIsRetrieved(String expiryInHours) {
+        mockDevelocityServer = EmbeddedApp.fromHandlers(
+                c -> c.post("api/auth/token", ctx -> ctx.getResponse().status(200).send(RandomStringUtils.randomAlphanumeric(50)))
+        );
+
+        DevelocityAccessCredential develocityAccessCredential = shortLivedTokenClient.get(
+                mockDevelocityServer.getAddress().toString(),
+                DevelocityAccessCredential.parse("localhost=" + RandomStringUtils.randomAlphanumeric(30), "localhost").get(),
+                expiryInHours
+        ).orElseThrow(() -> new IllegalStateException("Short lived token value is expected"));
+
+        assertThat(develocityAccessCredential.getHostname(), equalTo("localhost"));
+        assertThat(develocityAccessCredential.getKey(), not(isEmptyOrNullString()));
```

Comment:
I would use fixed string so you can assert on the expected short lived token value, here it could pass with the access key being set

Reply:
Agree, replaced